In [3]:
import numpy as np
import pandas as pd

### 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [5]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Найдём ковариацию с помощью элементарных действий:

In [12]:
cov_zpks = sum(zp*ks)/len(zp) - (sum(zp)/len(zp))*(sum(ks)/len(ks))
cov_zpks

9157.839999999997

Найдём ковариацию с помощью ф-ции cov (смещённая оценка):

In [17]:
np.cov(zp, ks, ddof=0)[0,1]

9157.84

Найдём коэф-т корреляции Пирсона:

In [20]:
cor_zpks = cov_zpks/(np.std(zp)*np.std(ks))
cor_zpks

0.8874900920739158

То же самое, но с помощью numpy:

In [23]:
np.corrcoef(zp,ks)[0, 1]

0.8874900920739162

С помощью pandas:

In [51]:
df = pd.DataFrame(zp,ks).reset_index()
df

index    0
0    401   35
1    574   45
2    874  190
3    919  200
4    459   40
5    739   70
6    653   54
7    902  150
8    746  120
9    832  110

In [54]:
df.corr().values[0,1]

0.8874900920739162

### 2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [24]:
X = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
mean_X = X.mean()
std_X = X.std(ddof=1)
t = 2.2622

In [25]:
a = mean_X - t*std_X/np.sqrt(len(X))
a

110.55594079648245

In [26]:
b = mean_X + t*std_X/np.sqrt(len(X))
b

125.64405920351754

Ответ: [110.56, 125.64]

### 3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с альфа =0,95

In [27]:
X = 174.2
Z = 1.96
σ = np.sqrt(25)
n = 27

In [28]:
a = X - Z*σ/np.sqrt(n)
a

172.31398912064722

In [29]:
b = X + Z*σ/np.sqrt(n)
b

176.08601087935276

Ответ: [172.31, 176.09]